Dockerfile

```diff
@@ -1,18 +1,32 @@
 FROM python:3.12-slim
 
-# Install GNU Parallel and other dependencies
+# Install GNU Parallel for benchmark runs
 RUN apt-get update && apt-get install -y parallel
 
-# Copy the entire project into the container
-COPY . /ucc
+# Install Poetry
+RUN pip3 install poetry==2.0.1
```

Comment:
Any reason not to use the "[official installer](https://python-poetry.org/docs/#installing-with-the-official-installer)" here?

Reply:
> avoid needing to apt-get curl in the dockerfile

ahh yeah true. I guess if it works, then 🤷🏻. I was just hesitant about using `pip` to install `poetry` since it's not on their docs, but I guess `pipx` (which is documented) is just `pip` with some fancy virtual env management.